homeassistant/components/ohme/binary_sensor.py

```diff
@@ -0,0 +1,90 @@
+"""Platform for binary_sensor."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import OhmeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+BINARY_SENSOR_DESCRIPTIONS = [
+    BinarySensorEntityDescription(
+        key="car_connected",
+        icon="mdi:ev-plug-type2",
+        device_class=BinarySensorDeviceClass.PLUG,
+    ),
+    BinarySensorEntityDescription(
+        key="car_charging",
+        icon="mdi:battery-charging-100",
+        device_class=BinarySensorDeviceClass.BATTERY_CHARGING,
+    ),
```

Comment:
those three are probably mutually exclusive right? So wouldn't it make more sense to use a normal `sensor` of type `enum` instead?

Reply:
Indeed they are and the enum approach seems to match with other EVSE integrations.

Where would you put the logic  to determine the enum value? In the external API library or somewhere in the integration?

Reply:
Ideally in the library